pantheon-bundle/src/test/java/com/redhat/pantheon/model/api/SlingModelAdapterFactoryTest.java

```diff
@@ -0,0 +1,92 @@
+package com.redhat.pantheon.model.api;
+
+import com.redhat.pantheon.model.Product;
+import com.redhat.pantheon.model.ProductVersion;
+import com.redhat.pantheon.model.api.v2.FileResource;
+import com.redhat.pantheon.model.api.v2.SlingModel;
+import com.redhat.pantheon.model.api.v2.SlingModelAdapterFactory;
+import com.redhat.pantheon.model.module.Content;
+import com.redhat.pantheon.model.module.Metadata;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.model.module.ModuleLocale;
+import com.redhat.pantheon.model.module.ModuleVersion;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Calendar;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class SlingModelAdapterFactoryTest {
+
+    private final SlingContext slingContext = new SlingContext();
+
+    SlingModelAdapterFactory modelAdapterFactory = new SlingModelAdapterFactory();
+
+    @Test
+    void getAdapter() {
+        // Given
+        slingContext.build()
+                .resource("/content/module1/draft/metadata",
+                        "jcr:createdBy", "a-user",
+                        "jcr:created", Calendar.getInstance())
+                .commit();
+
+        // When
+        ModuleVersion adapter = modelAdapterFactory.getAdapter(slingContext.resourceResolver().getResource("/content/module1/draft"),
+                ModuleVersion.class);
+
+        // Then
+        assertNotNull(adapter);
+        assertEquals("a-user", adapter.metadata().get().createdBy().get());
+        assertNotNull(adapter.metadata().get().created().get());
+    }
+
+    // Make sure there are no regressions that make adapters unavailable
+    @Test
+    void availableAdapters() {
+        // Given
+        Resource mockResource = Mockito.mock(Resource.class);
+
+        // Then
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, SlingModel.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Module.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ModuleVersion.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Content.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Metadata.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ModuleLocale.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, Product.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, ProductVersion.class));
+        assertNotNull(modelAdapterFactory.getAdapter(mockResource, FileResource.class));
```

Comment:
The purpose is to check that we don't break compatibility of the objects that rely on being adapted. You are right in that it might require some manual maintenance to make sure newly added models get tested by this. We could do some clever reflection stuff to make sure any model interface is picked up automatically. Let me look at this; could I persuade you to consider this as an improvement? to not block the whole PR.

Reply:
You are right about something.... this test should have been failing as some of the models were not listed in the annotation.

Reply:
Ok, I see now... the test is not covering the effects of the annotation. It just checks that it doesn't blow up when we pass it one of the given interfaces. It's not really giving us a lot.

Reply:
Yeah... I think I agree. The test has nothing to do with the annotation. You are calling the method directly, and you essentially have an 'if' statement that says 'if the requested type doesn't extend from SlingModel then fail, otherwise cast this thing to that type' (grossly oversimplifying here). And you are testing to make sure it didn't take the fail path.

Bottom line, let's leave the test as-is. It's not hurting anything, but also not worth trying to work some java reflection magic, since the annotation is not at play whatsoever.

Reply:
:+1: